src/diagnostics.rs

```diff
@@ -36,7 +36,7 @@ impl std::str::FromStr for Level {
             "HELP" | "help" => Ok(Self::Help),
             "NOTE" | "note" => Ok(Self::Note),
             "failure-note" => Ok(Self::FailureNote),
-            "error: internal compiler error" => Ok(Self::Ice),
+            "ICE" | "ice" | "error: internal compiler error" => Ok(Self::Ice),
```

Comment:
I'm not sure if it's supposed to be like this but we can never have `"error: internal compiler error"` here as far as I know since the content after the `:` is removed when parsing the pattern. So instead, I think it makes sense to add a new `ICE` level to handle this case.

Reply:
If there are no affected tests, yes.

Also add a test for the new annotation. There should be an existing Ztreat-err-as-bug test. If not, add one

Reply:
I was sure there was a `//@ failure-status: 101` command but apparently not. Might be worth as a follow-up. Anyway, added the test and removed the extra text matching.

Reply:
> was sure there was a `//@ failure-status: 101` command bu

Yea I should probably add that